profiling/src/profiling/stalk_walking.rs

```diff
@@ -2,14 +2,43 @@ use crate::bindings::{
     ddog_php_prof_zend_string_view, zend_execute_data, zend_function, zend_string,
     ZEND_USER_FUNCTION,
 };
+use crate::string_table::{OwnedStringTable, StringTable};
+use std::borrow::Cow;
+use std::cell::{RefCell, RefMut};
+use std::mem::transmute;
 use std::str::Utf8Error;
 
+/// Used to help track the function run_time_cache hit rate. It glosses over
+/// the fact that there are two cache slots used, and they don't have to be in
+/// sync. However, they usually are, so we simplify.
+#[cfg(php8)]
+#[derive(Debug, Default)]
+pub struct FunctionRunTimeCacheStats {
+    hit: usize,
+    missed: usize,
+    not_applicable: usize,
+}
+
+#[cfg(php8)]
+impl FunctionRunTimeCacheStats {
+    pub fn hit_rate(&self) -> f64 {
+        let denominator = (self.hit + self.missed + self.not_applicable) as f64;
+        self.hit as f64 / denominator
+    }
+}
+
+#[cfg(php8)]
+thread_local! {
+    static CACHED_STRINGS: RefCell<OwnedStringTable> = RefCell::new(OwnedStringTable::new());
+    pub static FUNCTION_CACHE_STATS: RefCell<FunctionRunTimeCacheStats> = RefCell::new(Default::default())
+}
+
 #[derive(Default, Debug)]
 pub struct ZendFrame {
     // Most tools don't like frames that don't have function names, so use a
-    // fake name if you need to like "<php>".
-    pub function: String,
-    pub file: Option<String>,
+    // fake name if you need to like "<?php".
+    pub function: Cow<'static, str>,
```

Comment:
Mostly reading to see if I understand well: So in some cases we want to own the string and give it static lifetime (What are these cases?), otherwise we will use the borrowed lifetime from the cache.